backend/inspirehep/refextract/views.py

```diff
@@ -0,0 +1,66 @@
+from flask import Blueprint, request
+from refextract import (
+    extract_journal_reference,
+    extract_references_from_file,
+    extract_references_from_string,
+)
+
+from inspirehep.records.utils import download_file_from_url
+from inspirehep.serializers import jsonify
+
+from .utils import get_journal_kb_dict
+
+blueprint = Blueprint("inspirehep_refextract", __name__, url_prefix="/refextract")
+
+
+@blueprint.route("/extract_journal_info", methods=["POST"])
+def extract_journal_info():
+    publication_infos = request.json["publication_info"]
+    extracted_publication_infos = []
+    journal_dict = {"journals": get_journal_kb_dict()}
+    try:
+        for publication_info in publication_infos:
+            extracted_publication_info = extract_journal_reference(
+                publication_info["pubinfo_freetext"],
+                override_kbs_files=journal_dict,
+            )
+            if not extracted_publication_info:
+                extracted_publication_info = {}
+            extracted_publication_infos.append(extracted_publication_info)
+    except TimeoutError:
```

Comment:
let's treat all the errors the same way

Reply:
one issue I see with this approach: refextract for papers with many refs will exceed the HTTP request timeout of `hep` (it can take 5 min sometimes). Also it will put more load on `hep` during the harvest (blocking web workers which are running refextract and are not available to serve user requests).